Server/src/services/db/log.controller.ts

```diff
@@ -33,28 +24,26 @@ export default class LogController {
      * Saves a new log in the database.
      *
      * @param timestamp - Time of log.
-     * @param trackerId - Tracker.uid which caused the log.
      * @param position - Current GPS position at the time of the creation of the log.
      * @param heading - Current GPS heading at the time of the creation of the log.
      * @param speed - Current speed at the time of the creation of the log.
-     * @param battery - Current battery charge at the time of the creation of the log.
+     * @param battery - Current battery charge at the time of the creation of the log. (Optional for the case of app data)
+     * @param trackerId - Tracker.uid which caused the log. (Optional for the case of app data)
      * @param data - optional addtional data field.
      * @returns Log | null if an error occurs.
      */
-    public async save(timestamp : Date, trackerId: string, position: JSON, heading: number, speed: number, battery: number, data?: JSON) : Promise<Log | null> {
+    public async save(timestamp : Date, vehicleId: number, position: JSON, heading: number, speed: number, battery?: number, data?: JSON, trackerId?: string) : Promise<Log | null> {
         try {
-            // TODO: vvv This.
-            let pos = JSON.parse(JSON.stringify(position)) as Prisma.InputJsonObject
-            let d = (data === undefined ? Prisma.JsonNull : JSON.parse(JSON.stringify(data))) as Prisma.InputJsonObject
             return await this.prisma.log.create({
                 data : {
                     timestamp: timestamp,
                     trackerId: trackerId,
-                    position: pos,
+                    position: (position as unknown as Prisma.InputJsonValue),

```

Comment:
Why does such a conversion work here, is that always the case? For unusual casts there should be some reasoning found in a comment about that.